### PR TITLE
gas needle with different motors and encoders installed

### PR DESCRIPTION
**Pull request by tongju12**
TMO swap the motors for gas needle by smaller size and less current. 
Meanwhile, adding encoder into every axe. 

Tune the parameter, link to encoder, rewrite code in main. 
## Motivation and Context

Request from scientists in new MRCO chamber

## How Has This Been Tested?

Yes, I tested with scientists to setup soft limits and move to the whole range for every axe.

## Where Has This Been Documented?

https://jira.slac.stanford.edu/browse/ECS-5513

## Pre-merge checklist
- [ ] Code works interactively
- [ ] Code contains descriptive comments
- [ ] Test suite passes locally
- [ ] Libraries are set to fixed versions and not ``Always Newest``
- [ ] Code committed with ``pre-commit`` (alternatively ``pre-commit run --all-files``)

### CODE REVIEW

**Comment by ZLLentz** [comment]
Tong, were you going to run pre-commit and push back here? I was going to review but was going to wait if you had more commits to push.

**Comment by tongju12** [comment]
Yes, Zach. Sorry. I am busy with Sample paddle X, Y, Z test today and it is almost done except Y motor has strange noise. I will push again with pre-commit run next week. Thanks!

**Comment by tongju12** [comment]
Zach, I ran pre-commit before second PR and hope it is ok for you?

File: lcls-plc-mrco-motion/_Config/NC/Axes/GasNozzleX.xti [hunk 1]
```diff
@@ -1328,12 +1328,11 @@ External Setpoint Generation:
...
-			<EncPara ScaleFactorNumerator="9.921875e-05" MaxCount="#x0000ffff">
-				<SoftEndMinControl Enable="true" Range="-100"/>
-				<SoftEndMaxControl Enable="true" Range="100"/>
-				<Inc RefMode="2" RefSoftSyncMask="#x0000ffff"/>
-				<ParameterChanged>8</ParameterChanged>
+		<Encoder Name="Enc" EncType="19">
+			<EncPara ScaleFactorNumerator="5e-05" MaxCount="#xffffffff">
+				<SoftEndMinControl Enable="true" Range="31"/>
+				<SoftEndMaxControl Enable="true" Range="52"/>
+				<Inc RefSoftSyncMask="#x0000ffff"/>
```
Comment (ZLLentz) [inline lcls-plc-mrco-motion/_Config/NC/Axes/GasNozzleX.xti:1335]: Wow, the stages changed a lot!

**Review by ZLLentz — APPROVED** [review]
Looks good to me overall! I'm hitting approve!

Small nitpick: you have a few commented out sections that should probably be deleted
It looks like they completely changed the hardware here
I didn't notice anything strange in the diff

**Pull request merged**